src/hivpy/resistance_mutations.py

```diff
@@ -422,38 +422,34 @@ def update_new_mutations_arising_art(self, pop: Population, sub_pop):
                                        self.calc_rttams_outcomes, sub_pop=possible_mutation_pop)
             pop.set_present_variable(col.RTTA_MUTATIONS, tams, possible_mutation_pop)
 
-            # rt184m
-            m184 = pop.transform_group([col.ON_3TC, col.ON_ISL, col.RT184_MUTATION],
-                                       self.calc_rt184m_outcomes, sub_pop=possible_mutation_pop)
-            if len(pop.apply_bool_mask(m184, possible_mutation_pop)) > 0:
-                pop.set_present_variable(col.RT184_MUTATION, MutationStatus.Majority,
-                                         pop.apply_bool_mask(m184, possible_mutation_pop))
-
-            # rt151m
-            q151 = pop.transform_group([col.ON_ZDV, col.RT151_MUTATION],
-                                       self.calc_rt151m_outcomes, sub_pop=possible_mutation_pop)
-            if len(pop.apply_bool_mask(q151, possible_mutation_pop)) > 0:
-                pop.set_present_variable(col.RT151_MUTATION, MutationStatus.Majority,
-                                         pop.apply_bool_mask(q151, possible_mutation_pop))
-
-            # rt65m
-            k65 = pop.transform_group([col.ON_TEN, col.ON_ZDV, col.RT65_MUTATION],
-                                      self.calc_rt65m_outcomes, sub_pop=possible_mutation_pop)
-            if len(pop.apply_bool_mask(k65, possible_mutation_pop)) > 0:
-                pop.set_present_variable(col.RT65_MUTATION, MutationStatus.Majority,
-                                         pop.apply_bool_mask(k65, possible_mutation_pop))
-
-            # rt103m, rt181m, and rt190m (nnrti mutations)
-            k103 = pop.transform_group([col.ON_NEV, col.ON_EFA,
-                                        col.RT181_MUTATION, col.RT190_MUTATION],
+            # calculate and set a new majority mutation
+            def set_new_majority_mutation(mutation_col: str, drug_cols: list[str], calc_func: function):
+                # find people undergoing a given mutation this time step
+                mutation_mask = pop.transform_group(drug_cols, calc_func, sub_pop=possible_mutation_pop)
+                mutation_pop = pop.apply_bool_mask(mutation_mask, possible_mutation_pop)
+                # set outcomes
+                if len(mutation_pop) > 0:
+                    pop.set_present_variable(mutation_col, MutationStatus.Majority, mutation_pop)
+
+            # m184
+            set_new_majority_mutation(col.RT184_MUTATION, [col.ON_3TC, col.ON_ISL, col.RT184_MUTATION],
+                                      self.calc_rt184m_outcomes)
+            # q151
+            set_new_majority_mutation(col.RT151_MUTATION, [col.ON_ZDV, col.RT151_MUTATION],
+                                      self.calc_rt151m_outcomes)
+            # k65
+            set_new_majority_mutation(col.RT65_MUTATION, [col.ON_TEN, col.ON_ZDV, col.RT65_MUTATION],
+                                      self.calc_rt65m_outcomes)
```

Comment:
Is there a reason these three mutations check for their own majority presence when none of the other mutations do?